lib/oli_web/live/delivery/student/lesson/components/one_at_a_time_question.ex

```diff
@@ -300,6 +351,52 @@ defmodule OliWeb.Delivery.Student.Lesson.Components.OneAtATimeQuestion do
     end
   end
 
+  def handle_event("select_question", %{"question_number" => question_number}, socket) do
+    questions =
+      socket.assigns.questions
+      |> Enum.map(fn question ->
+        Map.put(question, :selected, question.number == question_number)
+      end)
+
+    {:noreply, assign(socket, questions: questions)}
+  end
+
+  def handle_event(
+        "submit_selected_question",
+        %{"attempt_guid" => attempt_guid, "question_id" => question_id},
+        socket
+      ) do
+    ## evaluate the activity attempt
+
+    Oli.Repo.get_by(Oli.Delivery.Attempts.Core.ActivityAttempt,
+      attempt_guid: attempt_guid
+    )
+    |> Oli.Repo.preload([:resource_attempt, :part_attempts, :revision])
```

Comment:
I wonder if we could use or extend `Oli.Delivery.Attempts.Core.get_activity_attempt_by/1` so we avoid performing db queries in a component.